### PR TITLE
Cleanup: Move single lines from readonly section source to header

### PR DESCRIPTION
**Pull request by eleftherioszisis**

### CODE REVIEW

File: include/morphio/section.h [hunk 1]
```diff
@@ -42,46 +42,66 @@ class Section: public SectionBase<Section>
     /**
```
Comment (mgeplf) [inline include/morphio/section.h:42]: can you also collapse the single line comments?

Comment (eleftherioszisis) [reply include/morphio/section.h]: Done

File: include/morphio/section.h [hunk 2]
```diff
@@ -39,29 +39,23 @@ class Section: public SectionBase<Section>
     using PointAttribute = Property::Point;
 
   public:
-    /**
-       Depth first search iterator
-    **/
+    /// Depth firt search iterator
```
Comment (mgeplf) [inline include/morphio/section.h:42]: firt -> first

It's not really a `search` iterator, so we can drop that for all of them

Comment (eleftherioszisis) [reply include/morphio/section.h]: yeah, fixed

**Comment by eleftherioszisis** [comment]
Is this good to go?

**Review by mgeplf — APPROVED** [review]

**Pull request merged**